src/packages/hasura/hasura-events.ts

```diff
@@ -0,0 +1,185 @@
+      headers,
+      method: 'POST',
+    });
+
+    // Awaiting all requests in parallel
+    const [activityTypeResp, modelParameterResp, resourceTypeResp] = await Promise.all([
+      refreshActivityTypes
+        .then(response => {
+          if (!response.ok) {
+            logger.error(
+              `Bad status received when extracting model activity types: [${response.status}] ${response.statusText}`,
+            );
+            return {
+              error: `Bad status received when extracting model activity types: [${response.status}] ${response.statusText}`,
+              status: response.status,
+              statusText: response.statusText,
+            };
+          }
+          return response.json();
+        })
+        .catch(error => {
+          logger.error(`Error connecting to Hasura metadata API at ${metadataURL}. Full error below:\n${error}`);
+          return { error: `Error connecting to metadata API at ${metadataURL}` };
+        }),
+      refreshModelParameters
+        .then(response => {
+          if (!response.ok) {
+            logger.error(
+              `Bad status received when extracting model parameters: [${response.status}] ${response.statusText}`,
+            );
+            return {
+              error: `Bad status received when extracting model parameters: [${response.status}] ${response.statusText}`,
+              status: response.status,
+              statusText: response.statusText,
+            };
+          }
+          return response.json();
+        })
+        .catch(error => {
+          logger.error(`Error connecting to Hasura metadata API at ${metadataURL}. Full error below:\n${error}`);
+          return { error: `Error connecting to metadata API at ${metadataURL}` };
+        }),
+      refreshResourceTypes
+        .then(response => {
+          if (!response.ok) {
+            logger.error(
+              `Bad status received when extracting model resource types: [${response.status}] ${response.statusText}`,
+            );
+            return {
+              error: `Bad status received when extracting model resource types: [${response.status}] ${response.statusText}`,
+              status: response.status,
+              statusText: response.statusText,
+            };
+          }
+          return response.json();
+        })
+        .catch(error => {
+          logger.error(`Error connecting to Hasura metadata API at ${metadataURL}. Full error below:\n${error}`);
+          return { error: `Error connecting to metadata API at ${metadataURL}` };
+        }),
+    ]);
+
+    logger.info(`POST /modelExtraction: Extraction triggered for model: ${missionModelId}`);
+
+    res.json({
+      message: `Extraction triggered for model: ${missionModelId}`,
+      response: {
+        activity_types: await activityTypeResp,
+        model_parameters: await modelParameterResp,
+        resource_types: await resourceTypeResp,
+      },
+    });
+  });
+};
```

Comment:
Non-blocking: Refactoring to something like the following clears up the logic a little bit here, IMO

```typescript
  const generateBody = (name) => JSON.stringify({
    args: {
      name: `refresh${name}`,
      payload: { id: missionModelId },
      source: 'Aerie',
    },
    type: 'pg_invoke_event_trigger',
  });

  const refreshMetadata = async (name) => {
    try {
      const response = await fetch(metadataURL, {
        method: 'POST',
        headers,
        body: generateBody(name),
      });

      if (!response.ok) {
        logger.error(`Bad status received when extracting model ${name}: [${response.status}] ${response.statusText}`);
        return {
          error: `Bad status received when extracting model ${name}: [${response.status}] ${response.statusText}`,
          status: response.status,
          statusText: response.statusText,
        };
      }

      return response.json();
    } catch (error) {
      logger.error(`Error connecting to Hasura metadata API at ${metadataURL}. Full error below:\n${error}`);
      return { error: `Error connecting to metadata API at ${metadataURL}` };
    }
  };

  const [activityTypeResp, modelParameterResp, resourceTypeResp] = await Promise.all([
    refreshMetadata('ActivityTypes'),
    refreshMetadata('ModelParameters'),
    refreshMetadata('ResourceTypes'),
  ]);

  logger.info(`POST /modelExtraction: Extraction triggered for model: ${missionModelId}`);

  res.json({
    message: `Extraction triggered for model: ${missionModelId}`,
    response: {
      activity_types: activityTypeResp,
      model_parameters: modelParameterResp,
      resource_types: resourceTypeResp,
    },
  });
});
```

Reply:
I kept the `.then()` `.catch()` syntax and used a different name (`extract`), but otherwise took this suggestion, thanks!